Source/WebKit/WebProcess/Plugins/PDF/UnifiedPDF/UnifiedPDFPlugin.mm

```diff
@@ -1213,12 +1248,21 @@ static AffineTransform documentSpaceToPageSpaceTransform(const IntDegrees& pageR
     }
 }
 
+auto UnifiedPDFPlugin::toContextMenuItemTag(int tagValue) const -> ContextMenuItemTag
+{
+    static constexpr std::array regularContextMenuItemTags { ContextMenuItemTag::OpenWithPreview, ContextMenuItemTag::SinglePage, ContextMenuItemTag::SinglePageContinuous, ContextMenuItemTag::TwoPages, ContextMenuItemTag::TwoPagesContinuous, ContextMenuItemTag::ZoomIn, ContextMenuItemTag::ZoomOut, ContextMenuItemTag::ActualSize };
+    const auto isKnownContextMenuItemTag = WTF::anyOf(regularContextMenuItemTags, [tagValue](ContextMenuItemTag tag) {
+        return tagValue == enumToUnderlyingType(tag);
+    });
+    return isKnownContextMenuItemTag ? static_cast<ContextMenuItemTag>(tagValue) : ContextMenuItemTag::Unknown;
+}
+
 PDFContextMenu UnifiedPDFPlugin::createContextMenu(const IntPoint& contextMenuPoint) const
```

Comment:
I had to go look to see what type `PDFContextMenu` was, to check the memory safety. It's not a menu! It's more like `PDFContextMenuInfo` or `PDFContextMenuData`

Reply:
Yes `PDFContextMenu` and `NSMenu` are quite different object types. Retaining the `NSMenu` should be enough for memory safety!